Rethink `should_panic` and `fail_uncoverable` options as global conditions (#2967)
model-checking/kani: rfc/src/rfcs/0003-cover-statement.md

}
```

> The `--fail-uncoverable` option will allow users to fail the verification if a cover property is unsatisfiable or unreachable.

Author: Looks like the @adpaco-aws from the past already took that into account so the draft implementation in #2532 includes test cases for both and, when `--fail-uncoverable` is enabled, they fail as expected. That's because the current implementation checks that all `cover` properties are satisfiable for the `fail_uncoverable` condition to pass. Any other values in `cover` properties will result in failure.

> The `--fail-uncoverable` option will allow users to fail the verification if a cover property is unsatisfiable or unreachable.

Author: What I don't like so much about the draft implementation are the words being used. For example, there is a test in `fail-uncoverable/uncoverable/test.rs` with the description ``Checks that the `--fail-uncoverable` option causes a failure due to a cover statement not being covered.``. I don't like the terms `uncoverable` and `being covered`  so much because they're ambiguous (does `covered` mean reachable or satisfiable?). I'll make a revision later today.

This option will be integrated within the framework of [Global Conditions](https://model-checking.github.io/kani/rfc/rfcs/0007-global-conditions.html), which is used to define properties that depend on other properties.

Using the `--fail-uncoverable` option will enable the global condition with name `fail_uncoverable`.
Following the format for global conditions, the outcome will be one of the following:

>  1. `` - fail_uncoverable: FAILURE (encountered one or more cover statements which were not satisfied)``

Reviewer: I like your use of "expected" in `should_panic` below; IMHO using it here would make this clearer.

```suggestion
 1. `` - fail_uncoverable: FAILURE (expected to reach all cover statements, but one or more were not reached)``
```

Reviewer: Yes, much better. Thank you!
